### ISSUE TITLE
Table service does not snap source location reliably

### ISSUE THREAD

**Issue by ulicious** [Project-OSRM/osrm-backend#6588]
I am using the table service of the osrm project (web requests, no local OSRM) to handle several destinations at once. Problem is that I am getting several 0 distances between the source and the different destinations. I have checked the source of each request and it seems that sometimes the source does snap somewhere close to the Portuguese coast instead of the location where it should. But this behaviour seems to be very random.

For example, these two request work fine.
http://router.project-osrm.org/table/v1/driving/-69.9236,12.4375;-62.1009,-38.8944?sources=0&annotations=distance http://router.project-osrm.org/table/v1/driving/-69.9236,12.4375;-62.3005,-38.7831?sources=0&annotations=distance

Request return of the request with single destinations
![correct](https://user-images.githubusercontent.com/56553461/229132516-1af28672-e8bb-421f-be97-14b3757a0cf5.png)

---
But if I combine both destinations in a single request, I get the above described problem.
http://router.project-osrm.org/table/v1/driving/-69.9236,12.4375;115.7381,-32.0475;151.2095,-33.9731?sources=0&annotations=distance

Request return with combined destinations
![incorrect](https://user-images.githubusercontent.com/56553461/229132491-bdb725ab-0124-41d1-b54b-8a2128ef3613.png)

It seems like it's not a problem based on the destinations as both single destinations individually work fine. I have also tried to combine these two destinations with other destinations and this works fine again.

**Comment by murr999888** [comment]
http://router.project-osrm.org/table/v1/driving/-69.9236,12.4375;115.7381,-32.0475;151.2095,-33.9731?sources=0&annotations=distance

-69.9236,12.4375 - Aruba, island in Carribean sea
115.7381,-32.0475 - West coast of Australia
151.2095,-33.9731 - East coast of Australia
Are you sure there are routes between these points?

**Comment by ulicious** [comment]
Yes, there is no route between these locations. But in IMO this should be determined from OSRM. For example, I have following request:

http://router.project-osrm.org/table/v1/driving/-69.9236,12.4375;-61.8523,17.1228;115.7381,-32.0475;151.2095,-33.9731;151.2842,-23.8511;2.9337,51.23?sources=0&annotations=distance

Between all these locations shouldn't be a route as the source is an island disconnected from all destinations. Still, I get following result:

![dest1](https://user-images.githubusercontent.com/56553461/231382229-55cccb72-ab79-47a5-9308-f20cf3eb73da.PNG)

What you can see:
- The source is not snapped correctly. It's more than 4000km away from the given location
- Even though there shouldn't be a route between the source and the first destination, the result returns 0 as distance.
- All other routes are correct as the result is None when there is not route

And here is where it gets random. If I use only the source and the first destination, I get the correct result:

http://router.project-osrm.org/table/v1/driving/-69.9236,12.4375;-61.8523,17.1228?sources=0&annotations=distance

![dest2](https://user-images.githubusercontent.com/56553461/231383133-c5ef91ff-b8e4-44df-b210-38cd97488d4c.PNG)

The source is snapped correctly with Aruba and the distance to the destination is None.
